src/pytorch_lightning/loggers/logger.py

```diff
@@ -66,84 +39,12 @@ def after_save_checkpoint(self, checkpoint_callback: ModelCheckpoint) -> None:
         """
         pass
 
-    @abstractmethod
-    def log_metrics(self, metrics: Dict[str, float], step: Optional[int] = None) -> None:
-        """Records metrics. This method logs metrics as soon as it received them.
-
-        Args:
-            metrics: Dictionary with metric names as keys and measured quantities as values
-            step: Step number at which the metrics should be recorded
-        """
-        pass
-
-    @abstractmethod
-    def log_hyperparams(self, params: Union[Dict[str, Any], Namespace], *args: Any, **kwargs: Any) -> None:
-        """Record hyperparameters.
-
-        Args:
-            params: :class:`~argparse.Namespace` or `Dict` containing the hyperparameters
-            args: Optional positional arguments, depends on the specific logger being used
-            kwargs: Optional keyword arguments, depends on the specific logger being used
-        """
-
-    def log_graph(self, model: "pl.LightningModule", input_array: Optional[Tensor] = None) -> None:
-        """Record model graph.
-
-        Args:
-            model: lightning model
-            input_array: input passes to `model.forward`
-        """
-        pass
-
-    def save(self) -> None:
-        """Save log data."""
-
-    def finalize(self, status: str) -> None:
-        """Do any processing that is necessary to finalize an experiment.
-
-        Args:
-            status: Status that the experiment finished with (e.g. success, failed, aborted)
-        """
-        self.save()
-
     @property
     def save_dir(self) -> Optional[str]:
         """Return the root directory where experiment logs get saved, or `None` if the logger does not save data
         locally."""
         return None
 
-    @property
-    def group_separator(self) -> str:
-        """Return the default separator used by the logger to group the data into subfolders."""
-        return "/"
-
-    @property
-    @abstractmethod
-    def name(self) -> Optional[str]:
-        """Return the experiment name."""
-
-    @property
-    @abstractmethod
-    def version(self) -> Optional[Union[int, str]]:
-        """Return the experiment version."""
-
-
-class DummyExperiment:
-    """Dummy experiment."""
-
-    def nop(self, *args: Any, **kw: Any) -> None:
-        pass
-
-    def __getattr__(self, _: Any) -> Callable:
-        return self.nop
-
-    def __getitem__(self, idx: int) -> "DummyExperiment":
-        # enables self.logger.experiment[0].add_image(...)
-        return self
-
-    def __setitem__(self, *args: Any, **kwargs: Any) -> None:
-        pass
-
 
 class DummyLogger(Logger):
```

Comment:
Is this not relevant for Fabric?

Reply:
Not needed for now. If it's needed we will move it. Note: If I had moved it, another version of @carmocca would have said: "This isn't used anywhere in Fabric, so we should not have it there" :))

Reply:
You did the correct thing by not moving it. I was just asking if there were any future plans to use it in Fabric. I guess not because there's no concept of fast dev run

Reply:
Yes. One could build a trainer with fabric and have a fast_dev_run feature. In this case, this dummy logger would be a mere utility piece in Fabric.